operator/pkg/bgpv2/cell.go

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package bgpv2
+
+import (
+	"github.com/spf13/pflag"
+
+	"github.com/cilium/cilium/pkg/hive/cell"
+)
+
+const (
+	// BGPv2Enabled is the name of the flag that enables BGPv2 APIs in Cilium.
+	BGPv2Enabled = "enable-bgp-v2-api"
+)
+
+var Cell = cell.Module(
+	"bgp-cp-operator",
+	"BGP Control Plane Operator",
+	cell.Config(Config{}),
+	cell.Provide(NewBGPResourceManager),
+	cell.Invoke(func(*BGPResourceManager) {}),
```

Comment:
Just another suggestion: it seems that, for now, there aren't any other module that requires a `BGPResourceManager` (and I see that the only exported method, `Run`, is used only in this Module).
Unless you plan to use the manager externally in a follow-up PR, you can remove the Provide+Invoke and do something similar to, for example, `registerGC` in `operator/endpointgc/cell.go`.
Namely, you can call `cell.Invoke(registerBGPResourceManager)` in your `cell.Module` and append the manager obj to the passed in hive.Lifecycle (either implementing the `HookInterface` withing the manager, or using a `hive.Hook` struct).

Reply:
I removed Invoke from the cell. I am facing issues when moving when moving from Provide -> Invoke(registerBGPResourceManager). Externally tests are using BGPResourceManager - so if I do not do provide in the cell - tests are not able to use the manager. 

https://github.com/cilium/cilium/pull/28846/files#diff-dcebd6cffeee8a12ee4a2f0d485939a9d2eca0c715a46439a643ba3482b14905R121-R123

Reply:
That is not gonna work when the operator is run outside of tests.
Since there is no other consumer of the `*BGPResourceManager` type in the operator hive, you need the invoke to force the call to `NewBGPResourceManager` and register the manager jobs into the operator lifecycle.
Without that, the hive won't run `NewBGPResourceManager`, because the type *BGPResourceManager`, will be seen as not needed in the dependency graph.
Since we always want to register the BGP manager even if there are no other modules depending on it, you should keep the `cell.Invoke` and get rid of the `cell.Provide` instead.

Regarding the Invoke in `fixture_test.go`, you can simply remove it, because a reference to the manager is not needed (`f.manager` is unused in the tests). Adding `Cell` in the test hive is enough:

https://github.com/cilium/cilium/pull/28846/files#diff-dcebd6cffeee8a12ee4a2f0d485939a9d2eca0c715a46439a643ba3482b14905R126

because the `cell.Invoke` in it will register the BGP manager jobs into the test hive.

Have a look at [this commit](https://github.com/pippolo84/cilium/commit/25397ef0b4ce3840cc61cb5c65df450dea2e2706) to see an example of how to do that.

Reply:
Thanks @pippolo84, amazing feedback. Got it working - here is the change https://github.com/cilium/cilium/compare/3236250e27bc15d363349739651271928f3a911a..6c8cbeb98cbebf828bc06f764c53e4e9e2637fc5